fs-api-blockio/src/System/FS/BlockIO/API.hs

```diff
@@ -55,11 +55,26 @@ data HasBlockIO m h = HasBlockIO {
   , submitIO :: HasCallStack => V.Vector (IOOp (PrimState m) h) -> m (VU.Vector IOResult)
     -- | Retrieve the parameters that this 'HasBlockIO' was initialised with.
   , getParams :: !IOCtxParams
+    -- | Get the current value of the @NoCache@ flag from the file handle.
+    --
+    -- This has different effects on different distributions.
+    -- * [Linux]: get the value of the @O_DIRECT@ flag.
+    -- * [MacOS]: returns 'False' by default.
+    -- * [Windows]: returns 'False' by default.
+  , readNoCache :: Handle h -> m Bool
+    -- | Set the current value of the @NoCache@ flag on a file handle.
+    --
+    -- This has different effects on different distributions.
+    -- * [Linux]: set the @O_DIRECT@ flag.
+    -- * [MacOS]: set the @F_NOCACHE@ flag.
+    -- * [Windows]: no-op.
+  , writeNoCache :: Handle h -> Bool -> m ()
```

Comment:
Maybe these should be named `hReadNoCache` and `hWriteNoCache` instead

Reply:
I would have only one and call it `hSetNoCache`.

We don't ever need to read this value, and in any case it is not supported on OSX. I also can't imagine any other application where you'd really need to know dynamically and based just on the handle itself, if one is using no-caching mode.

So lets have a setter with no getter. And to match the API naming convention for other `System.IO` things we should call it `hSetFoo` for some value of Foo (see e.g. `hSetBuffering` etc.).